src/main/kotlin/dev/minn/jda/ktx/JDABuilder.kt

```diff
@@ -0,0 +1,399 @@
+        }
+
+    var eventManager: IEventManager? = null
+        set(value) {
+            builder.setEventManager(value)
+            field = value
+        }
+
+    var audioSendFactory: IAudioSendFactory? = null
+        set(value) {
+            builder.setAudioSendFactory(value)
+            field = value
+        }
+
+    var idle: Boolean = false
+        set(value) {
+            builder.setIdle(value)
+        }
+
+    var activity: Activity? = null
+        set(value) {
+            builder.setActivity(value)
+            field = value
+        }
+
+    var status: OnlineStatus? = null
+        set(value) {
+            if (value != null) {
+                builder.setStatus(value)
+            }
+            field = value
+        }
+
+    var eventListeners: MutableSet<Any> = mutableSetOf()
+
+    var maxReconnectDelay: Int = 900
+        set(value) {
+            builder.setMaxReconnectDelay(value)
+            field = value
+        }
+
+    inline fun sharding(build: InlineSharding.() -> Unit) {
+        val sharding = InlineSharding().also(build)
+        builder.useSharding(sharding.shardId, sharding.shardTotal)
+    }
+
+    var sessionController: SessionController? = null
+        set(value) {
+            builder.setSessionController(value)
+            field = value
+        }
+
+    var voiceDispatchInterceptor: VoiceDispatchInterceptor? = null
+        set(value) {
+            builder.setVoiceDispatchInterceptor(value)
+            field = value
+        }
+
+    var chunkingFilter: ChunkingFilter? = null
+        set(value) {
+            builder.setChunkingFilter(value)
+            field = value
+        }
+
+    var gatewayIntents: MutableSet<GatewayIntent> = mutableSetOf()
+
+    inline fun enableIntents(cacheFlag: GatewayIntent, vararg cacheFlags: GatewayIntent) {
+        builder.enableIntents(cacheFlag, *cacheFlags)
+    }
+
+    inline fun enableIntents(cacheFlags: Set<GatewayIntent>) {
+        builder.enableIntents(cacheFlags)
+    }
+
+    inline fun disableIntents(cacheFlag: GatewayIntent, vararg cacheFlags: GatewayIntent) {
+        builder.disableIntents(cacheFlag, *cacheFlags)
+    }
+
+    inline fun disableIntents(cacheFlags: Set<GatewayIntent>) {
+        builder.disableIntents(cacheFlags)
+    }
+
+    var largeThreshold: Int = 250
+        set(value) {
+            builder.setLargeThreshold(value)
+            field = value
+        }
+
+    var maxBufferSize: Int = 2048
+        set(value) {
+            builder.setMaxBufferSize(value)
+            field = value
+        }
+
+    var injectKtx: Boolean = true
+
+    class InlineSharding {
+        var shardId: Int = 0
+        var shardTotal: Int = 0
+    }
+
+    fun build(): JDA {
+        builder.addEventListeners(eventListeners)
+        if (gatewayIntents.isNotEmpty())
+            builder.setEnabledIntents(gatewayIntents)
+        if (cacheFlags.isNotEmpty())
+            builder.setEnabledCacheFlags(enumSetOf(*cacheFlags.toTypedArray()))
```

Comment:
This should not be here

Reply:
Where should it be moved to then? I added it because it's build logic for the wrapped builder, so it'd intuitively go in the build method.

Reply:
I mean that using these setters should be avoided in general and instead the `enableCache` and `enableIntents` methods should be used instead. The setters will replace the defaults which is a footgun.

Reply:
Yeah, I know.
They should probably be both annotated with `@Deprecated`, and have a warning in the javadoc saying not to use the `gatewayIntents` or `cacheFlags` variables. But I still added that in, because it's functionality that the builder has, and I don't want to restrict anything.